Transcendence/TransCore/RPGCode.xml

```diff
@@ -1742,6 +1742,39 @@
 					)
 				))
 
+			(setq rpgGetPlayerPrimaryCurrency (lambda ()
+				"Returns the type of currency that the player has the greatest monetary value of"
+				(block
+					(
+						(allCurrencies (typFind "$"))
+						(bestCurrency (unv@ 'defaultCurrency))
+						(bestCurrencyValue -1)
+						)
+
+					(enum allCurrencies theCurrency
+						(block (
+								(curCurrencyValue (ecoExchange (plyGetCredits gPlayer theCurrency) theCurrency &ecCreditEconomy;))
+								)
+							(if (gr curCurrencyValue bestCurrencyValue)
+								(block ()
+									(setq bestCurrency theCurrency)
+									(setq bestCurrencyValue curCurrencyValue)
+									)
+								)
+							)
+						)
+
+					(if	(= bestCurrencyValue 0)
+
+						;;	If the player has no money, return the universe default
+						(unv@ 'defaultCurrency)
+
+						;;	otherwise return the best currency
+						bestCurrency
+						)
+					)
```

Comment:
I think you can replace most of this with a single `map` call. (I can't test this at the moment - so you might need to add an exludeNil or `(or ... 0)` if plyGetCredits returns a Nil rather than zero. Alternatively if the map returns Nil when the player has no money replace the if check with (or bestCurrency (unv@ 'defaultCurrency))
```suggestion
					(
					(bestCurrency (map (typFind "$") '(reduceMax original) theCurrency
						(ecoExchange (plyGetCredits gPlayer theCurrency) theCurrency &ecCreditEconomy;)
						)
					)
					(if (= (plyGetCredits gPlayer bestCurrency) 0)

						;;	If the player has no money, return the universe default
						(unv@ 'defaultCurrency)

						;;	otherwise return the best currency
						bestCurrency
						)
					)
```

Reply:
You can do that?

map's options are not properly documented, nor is the ability to combine them

Reply:
Can you check that doc PR and see if we have all the important cases covered in the unit  tests?

Reply:
(I was replying to your comment about the poor documentation - I hadn't seen your comment about improving the documentation when I posted!)